src/rpc/rawtransaction.cpp

```diff
@@ -206,13 +214,31 @@ static RPCHelpMan getrawtransaction()
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, errmsg + ". Use gettransaction for wallet transactions.");
     }
 
-    if (!fVerbose) {
+    if (verbosity == 0) {
         return EncodeHexTx(*tx, RPCSerializationFlags());
     }
 
     UniValue result(UniValue::VOBJ);
-    if (blockindex) result.pushKV("in_active_chain", in_active_chain);
-    TxToJSON(*tx, hash_block, result, chainman.ActiveChainstate());
+    if (blockindex) {
+        result.pushKV("in_active_chain", in_active_chain);
+    }
+
+    std::optional<size_t> opt_tx_position{std::nullopt};
+    CBlockUndo blockUndo;
+    CBlock block;
+    if (verbosity == 1 || !blockindex || IsBlockPruned(blockindex) || !(UndoReadFromDisk(blockUndo, blockindex) && ReadBlockFromDisk(block, blockindex, Params().GetConsensus()))) {
+        TxToJSON(*tx, hash_block, result, chainman.ActiveChainstate());
+        return result;
+    }
+    for (size_t i = 0; i < block.vtx.size(); ++i) {
+        const CTransactionRef t = block.vtx.at(i);
+        if (*t == *tx) {
+            // blockundo does not have coinbase tx
+            opt_tx_position = std::optional<size_t>{i-1};
+            break;
```

Comment:
I'm just studying for the pr review club. 

 There was a question about the performance implications of this loop. I was digging through the code and I was perhaps thinking that CBlockUndo could maybe benefit from a faster data structure to find a transaction? Maybe a method that returns the data faster.

Perhaps that is a later improvement and off topic?

Reply:
Someone did suggest a more performant  solution, I can't find it at the moment though.